infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -284,16 +282,18 @@ private void prefixTrim(CorfuPayloadMsg<TrimRequest> msg, ChannelHandlerContext
     }
 
     @ServerHandler(type = CorfuMsgType.READ_REQUEST)
-    private void read(CorfuPayloadMsg<ReadRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
+    public void read(CorfuPayloadMsg<ReadRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
         long address = msg.getPayload().getAddress();
-        log.trace("read: {}", msg.getPayload().getAddress());
+        boolean cacheable = msg.getPayload().isCacheReadResult();
+        log.trace("read: {}, cacheable: {}", msg.getPayload().getAddress(), cacheable);
+
         ReadResponse rr = new ReadResponse();
         try {
-            ILogData e = dataCache.get(address);
-            if (e == null) {
+            ILogData ld = dataCache.get(address, cacheable);
+            if (ld == null) {
```

Comment:
you can hide this `if else` block inside dataCache.get(key, putFunction)`

Reply:
This if else logic that returns LogData.getEmpty() is part of the log unit server's logic, not the cache's responsibility and capability, so I think it's better to put outside the cache.